src/helpers/momentHelper.js

```diff
@@ -137,6 +137,196 @@ const getTodayWithTime = (time) => {
   return moment(time, 'HH:mm').toDate();
 };
 
+/**
+ * Parses an opening time into a time interval
+ * @param {{
+ *  open: boolean | null;
+ *  weekday: string;
+ *  dateFrom: string;
+ *  dateTo: string;
+ *  timeFrom: string;
+ *  timeTo: string
+ * }} openingTime
+ */
+const getIntervalFromOpeningTime = (openingTime) => {
+  return {
+    timeFrom: openingTime.timeFrom?.length ? getTodayWithTime(openingTime.timeFrom) : undefined,
+    timeTo: openingTime.timeTo?.length ? getTodayWithTime(openingTime.timeTo) : undefined
+  };
+};
+
+/**
+ * merges two time intervals, if they overlap
+ * @param {{
+ *  timeFrom?: Date;
+ *  timeTo?: Date;
+ * }} timeA
+ * @param {{
+ *  timeFrom?: Date;
+ *  timeTo?: Date;
+ * }} timeB
+ * @returns {{
+ *  timeFrom?: Date;
+ *  timeTo?: Date;
+ * } | undefined} the merged interval, if they overlapped, or undefined otherwise
+ */
+// eslint-disable-next-line complexity
+const mergeIntervals = (timeA, timeB) => {
+  // check which one starts first
+  let isAFirst = false;
+
+  if (timeA.timeFrom && timeB.timeFrom) {
+    isAFirst = timeA.timeFrom < timeB.timeFrom;
+  } else if (!timeA.timeFrom) {
+    isAFirst = true;
+  }
+  const first = isAFirst ? timeA : timeB;
+  const second = isAFirst ? timeB : timeA;
+
+  // check if the one that starts first ends after the other one started, which means that they do not overlap
+  if (first.timeTo && second.timeFrom && first.timeTo < second.timeFrom) {
```

Comment:
we would need something like `if(first.timeTo && moment(first.timeTo).isBefore(second.timeFrom))`.

we could not omit the check if `first.timeTo` is present because `moment(undefined)` will evaluate to the current time.

i do not see that as being more readable, and neither do i think it would make a comment less necessary.

Reply:
ok, using moment in this case is different but not better. we keep it as is.